File: JanperCab/3-Application/Dtos/Process/ProcessPackingDto.cs
using _3_Application.Dtos.Machine;

namespace _3_Application.Dtos.Process
{
    public class ProcessPackingDto : ProcessDto
    {
        public int? MachineId { get; set; }

        public MachinePackingDto MachinePacking { set; get; }
    }
}

File: JanperCab/5-JanperCab/ClientApp/src/app/_models/process/ProcessPackingDto.ts
import { MachinePackingDto } from '../machine/MachinePackingDto';
import { ProcessDto } from './ProcessDto';

export class ProcessPackingDto extends ProcessDto {
  machineId: number;

  machinePacking: MachinePackingDto;

  getStatus(): string {
    let status = this.startTime && this.endTime ? 'Packed' : 'Packing';

    if (this.machinePacking) {
      status += ` - ${this.machinePacking.name}`;
    }

    return status;
  }
}
